src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -51,15 +58,66 @@ const Zmenu = (props: ZmenuProps) => {
 
   const destroyZmenu = () => {
     dispatch(changeHighlightedTrackId(''));
-    setZmenus && setZmenus(pickBy(zmenus, (value, key) => key !== props.id));
+    setZmenus &&
+      setZmenus(pickBy(zmenus, (value, key) => key !== props.zmenuId));
   };
 
   const direction = chooseDirection(props);
   const toolboxPosition =
     direction === Direction.LEFT ? ToolboxPosition.LEFT : ToolboxPosition.RIGHT;
 
+  const { content, variety } = props.payload;
+  const features: (ZmenuContentTranscript | ZmenuContentGene)[] = [];
+
+  // get the first variety type
+
+  let featureId = '',
+    unversionedTranscriptId = '';
+
+  const zmenuType = variety.find((variety: ZmenuPayloadVariety) =>
+    Boolean(variety.type)
+  )?.type;
+
+  if (zmenuType === ZmenuPayloadVarietyType.GENE_AND_ONE_TRANSCRIPT) {
+    const transcript = content.find(
+      (feature: ZmenuContentTranscript | ZmenuContentGene) =>
+        feature.metadata.type === 'transcript'
+    ) as ZmenuContentTranscript;
+
+    if (!transcript) {
+      return null;
+    }
+
+    features.push(transcript);
+
+    const gene = content.find(
+      (feature: ZmenuContentTranscript | ZmenuContentGene) =>
+        feature.metadata.type === 'gene' &&
+        feature.metadata.id === transcript.metadata.gene_id
+    ) as ZmenuContentGene;
+
+    if (!gene) {
+      return null;
+    }
+
+    if (gene) {
+      features.push(gene);
+    }
+    // FIXME: we need this hack until ENSWBSITES-1375 is fixed
+    unversionedTranscriptId = transcript.metadata.transcript_id.split(
+      '.'
+    )[0] as string;
```

Comment:
Isn't this going to break some species (wheat?)? I thought some species' unversioned transcripts end in a dot and a number. I could have sworn that this is the case for wheat.

Reply:
I tried using versioned stable id and it worked.